src/compiler/scala/tools/nsc/symtab/classfile/Pickler.scala

```diff
@@ -124,14 +130,21 @@ abstract class Pickler extends SubComponent {
     override protected def shouldSkipThisPhaseForJava: Boolean = !settings.YpickleJava.value
   }
 
+  // temporary variables used to populate a pickle
+  // reuse is handled by Pickle#active: each pickle gets to use them until `finish` is called,
+  // after which access will throw an assertion error.
+  private var index   = new mutable.AnyRefMap[AnyRef, Int]
+  private var entries = new Array[AnyRef](256)
+
   private class Pickle(root: Symbol) extends PickleBuffer(new Array[Byte](4096), -1, 0) {
     private val rootName  = root.name.toTermName
     private val rootOwner = root.owner
-    private var entries   = new Array[AnyRef](256)
     private var ep        = 0
-    private val index     = new mutable.AnyRefMap[AnyRef, Int]
+    private var active    = true
     private lazy val nonClassRoot = findSymbol(root.ownersIterator)(!_.isClass)
 
+    def finish() { active = false; index.clear(); fill(entries, null) }
```

Comment:
I miss procedure syntax already.

Reply:
Oh shoot, I'm not supposed to say `) {` anymore.